File: prjs/prog/prblm/projecteuler/Prob57.cpp
#include<iostream>
using namespace std;
typedef unsigned long long ull;
ull gcd(ull m,ull n){
	ull tmp;
	while(n>0){
		tmp = m%n;
		m=n;
		n=tmp;
	}
	return m;
}
ull d(ull n){
	ull size=0;
	while(n>0){
		n=n/10;
		size++;
	}
	return size;
}
int main(){
	ull p=3,q=2,tp,tq;
	ull count=1;
	ull result=0;
	while(count<1001){
	    tp = p+q+q;
	    tq = p+q; 
            p = tp; q = tq;
	    tp = gcd(p,q);
	    p = p/tp;
	    q= q/tp;
	    if(d(p)>d(q)){
		    result++;
	    } 
	    cout<<p<<"/"<<q<<" ----- "<<result<<endl;
            count++;
	}
        cout<<result<<endl;

}

File: prjs/prog/prblm/projecteuler/Prob57.py
#!/usr/bin/python
import math
def gcd(m,n):
	
	while(n>0):
		tmp = m%n
		m=n
		n=tmp
	
	return m;

def d(n):
	size=0
	while(n>0):
		n=int(n/10)
		size=size+1
	
	return size


p=3
q=2
count=1;
result=0;
while(count<1001):
    tp = p+q+q
    tq = p+q
    p = tp
    q = tq
    tp = gcd(p,q);
    p = p/tp;
    q= q/tp;
    if(d(p)>d(q)):
	    result=result+1;
	     
    count=count+1;
print result


